mpas_analysis/shared/generalized_reader/generalized_reader.py

```diff
@@ -124,9 +130,35 @@ def open_multifile_dataset(fileNames, calendar, simulationStartTime=None,
                                  startDate=startDate,
                                  endDate=endDate)
 
-    ds = xarray.open_mfdataset(fileNames,
-                               preprocess=preprocess_partial,
-                               decode_times=False, concat_dim='Time', **kwargs)
+    # we don't want to decode times, since we're typically using dates
+    # outside the allowed range
+    kwargs['decode_times'] = False
+    # files differ along the Time dimension
+    kwargs['concat_dim'] = 'Time'
+
+    autoclose = ((autocloseFileCount is not None) and
+                 len(fileNames) > autocloseFileCount)
+
+    openedWithAutoclose = False
+    if autoclose:
+        try:
+            ds = xarray.open_mfdataset(fileNames,
+                                       preprocess=preprocess_partial,
+                                       autoclose=True, **kwargs)
```

Comment:
Note: I don't want to modify `kwargs` to include `autoclose` because I will use it later without `autoclose` if this fails.  If `autoclose` is included in `kwargs`, this line will fail and the warning below will likely be printed but then it will try again below without explicitly including `autoclose`, so I think it will work out.